src/index.ts

```diff
@@ -249,42 +232,40 @@ export function ruleFactory<
           stepRow,
           stepCount,
         });
-        break;
-      } else if ('try' in rule) {
+        // eslint-disable-next-line @typescript-eslint/no-throw-literal
+        throw BREAK;
```

Comment:
Maybe return something here instead? So errors don't get mixed in with a success path.